projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php

```diff
@@ -1074,11 +1074,27 @@ function get_paywall_blocks( $newsletter_access_level ) {
 		: __( 'Already a subscriber?', 'jetpack' );
 
 	$sign_in = '';
-	if ( ! is_user_logged_in() && ( new Host() )->is_wpcom_simple() ) {
+	$host    = new Host();
+	if ( ! is_user_logged_in() && $host->is_wpcom_simple() ) {
+		$id           = '';
 		$sign_in_link = wpcom_logmein_redirect_url( get_current_url(), false, null, 'link' );
+		$button_text  = esc_html__( 'Log in', 'jetpack' );
+	} elseif ( class_exists( 'Automattic\Jetpack\Extensions\Premium_Content\Subscription_Service\Jetpack_Token_Subscription_Service' ) &&
+		! $host->is_wpcom_simple() ) {
+		// We are on Jetpack and no cookie is set
+		$id              = 'jp_retrieve_subscriptions_link';
+		$sign_in_link    = '#'; // listening to "click" event in view.js
+		$access_question = __( 'I am already subscribed.', 'jetpack' );
+		if ( ! Jetpack_Token_Subscription_Service::has_token_from_cookie() ) {
+			$button_text = esc_html__( 'Log in', 'jetpack' );
+		} else {
+			$button_text = esc_html__( 'Switch accounts', 'jetpack' );
+		}
+	}
 
+	if ( ! empty( $sign_in_link ) ) {
```

Comment:
let's use the approach in https://github.com/Automattic/jetpack/pull/33695 as a fallback, it should result in a better UX on all platforms.

`$sign_in_link = 'https://subscribe.wordpress.com/memberships/jwt?site_id=' . \Jetpack_Options::get_option( 'id' ) . '&redirect_url=' . get_current_url();`

Reply:
What do you mean "as a fallback"?

Reply:
when `! is_user_logged_in()` and `empty( $sign_in_link )`, or that would never happen?

Reply:
Outside of WPCOM, `is_user_logged_in()` doesn't have any meaning, so 
```
! is_user_logged_in() && && $host->is_wpcom_simple()
``` 
is already covered and $sign_in_link is never empty,....

Reply:
> is already covered and $sign_in_link is never empty,....

why do we check for `if ( ! empty( $sign_in_link ) ) {` then?

Reply:
Because if you are logged in AND on WPCOM `$sign_in_link` is empty...

Reply:
> I think we should display `Switch accounts` in that case!

So linking to logout? Yes it could be an idea...